src/main/java/web/Calculator.java

```diff
@@ -0,0 +1,102 @@
+package web;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+class Calculator{
+    private static int calculate (int num1, int num2, char op) {
+        int number1 = num2;
+        int number2 = num1;
+        switch (op) {
+            case '*':
+                return number1 * number2;
+            case '/':
+                return number1 / number2;
+            case '+':
+                return number1 + number2;
+            case '-':
+                return number1 - number2;
+            default:
+                return 0;
+        }
+    }
+
+    private Map<String, String> variables;
+
+    Calculator(Map<String, String> variables) {
+        this.variables = variables;
+    }
+
+    int solve(String equation) {
+        Stack<Integer> resultStack = new Stack<>();
+        String finalEquation = toNormal(equation);
+        for (int i = 0; i < finalEquation.length(); i++){
+            char c = finalEquation.charAt(i);
+            if (Pattern.matches("[0-9]", Character.toString(c))) {
+                int number = Character.getNumericValue(c);
+                resultStack.push(number);
+            }
+            else if (isOperator(Character.toString(c)))
+                resultStack.push(calculate(resultStack.pop(), resultStack.pop(), c));
+            else if (variables.containsKey(String.valueOf(c)) || (Object) c instanceof Integer)
+                resultStack.push(placeVariable(c));
+        }
+        return resultStack.pop();
+    }
+
+    private int placeVariable(char c) {
+        String value = variables.get(String.valueOf(c));
+        if (Pattern.matches("[a-z]", Character.toString(value.charAt(0)))) {
+            while (value.charAt(0) >= 'a' && value.charAt(0) <= 'z'){
+                return placeVariable(value.charAt(0));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using a branching statement as the last in a loop.](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4846841)